### PR TITLE
secure database secrets and ensure middleware access

### PR DESCRIPTION
**Pull request by marycrawford**
## Description
Utilize infrastructure as code to automate the creation of the database user and secrets, as well as keeping it secure by storing it in Azure key vault.  Passing the values to the middleware to access (while still making sure security is added with the sslmode).

## Related Issues
[Link any related issues or tasks from your project management system.]

## Checklist
- [x] The title of this PR is descriptive and concise.
- [x] My changes follow the style guidelines of this project.
- [x] Considered an architectural design with a security first approach to keep the secret for the database safe.
- [x] I've let the team know about this PR by linking it in the review channel

### CODE REVIEW

File: ops/terraform/main.tf [hunk 1]
```diff
@@ -98,6 +98,10 @@ module "ocr_autoscale" {
   weekend_capacity_instances = 1
 }
 
+module "vault" {
+  location       = data.azurerm_resource_group.rg.location
```
Comment (derekadombek) [inline ops/terraform/main.tf:102]: We should pre-create/manually create vault secrets ahead and bringing them in as data sources similar to how we are with the Resource Group's, like this for example: https://github.com/CDCgov/prime-simplereport/blob/593c56c62847fbf398a0a8cddc5e4069a1499bd1/ops/prod/_data.tf#L50 This is also how CDC azure access will be. 

Unfortunately it looks like we might not have permissions to create key vault secrets at the moment in Azure. I ping'd Josh earlier so see if he can add a role to allow it.

Comment (marycrawford) [reply ops/terraform/main.tf]: We met with Josh Dorothy on Tuesday, November 26 and received permissions.  In that meeting you agreed on this architectural design.  In that meeting, I re-iterated the manual creation of the secret may not be the best practice or most secure.

File: dev-env.yaml [hunk 2]
```diff
@@ -30,9 +30,11 @@ services:
     ports:
       - "5432:5432"
     environment:
-      POSTGRES_DB: reportvision
-      POSTGRES_USER: postgres
```
Comment (arinkulshi-skylight) [inline dev-env.yaml:34]: We might need to make a separate app.yaml file but this is something we can handle as part of next steps. This looks ok!

**Review by arinkulshi-skylight — APPROVED** [review]
LG pending any changes from Derek!

**Pull request merged**